auth/api/iam/metadata.go

```diff
@@ -22,32 +22,73 @@ import (
 	"net/url"
 	"strings"
 
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/nuts-foundation/go-did/did"
 	"github.com/nuts-foundation/nuts-node/auth/oauth"
 	"github.com/nuts-foundation/nuts-node/core"
 	"github.com/nuts-foundation/nuts-node/crypto/jwx"
+	"github.com/nuts-foundation/nuts-node/vdr/didweb"
 )
 
-func authorizationServerMetadata(identity url.URL, oauth2BaseURL url.URL) oauth.AuthorizationServerMetadata {
+func authorizationServerMetadata(ownedDID did.DID) (*oauth.AuthorizationServerMetadata, error) {
+	if ownedDID.Method == "web" {
+		return _authzMetadataDidWeb(ownedDID)
+	}
+	return _authzMetadataBase(ownedDID), nil
+}
+
+func _authzMetadataDidWeb(ownedDID did.DID) (*oauth.AuthorizationServerMetadata, error) {
+	identity, err := didweb.DIDToURL(ownedDID)
+	if err != nil {
+		return nil, err
+	}
+	oauth2BaseURL, err := createOAuth2BaseURL(ownedDID)
+	if err != nil {
+		// can't fail, already did DIDToURL above
+		return nil, err
+	}
+	metadata := _authzMetadataBase(ownedDID)
+	metadata.Issuer = identity.String()
+	metadata.AuthorizationEndpoint = oauth2BaseURL.JoinPath("authorize").String()
+	metadata.PresentationDefinitionEndpoint = oauth2BaseURL.JoinPath("presentation_definition").String()
+	metadata.TokenEndpoint = oauth2BaseURL.JoinPath("token").String()
+	return metadata, nil
+}
+
+func _authzMetadataBase(ownedDID did.DID) *oauth.AuthorizationServerMetadata {
 	presentationDefinitionURISupported := true
-	return oauth.AuthorizationServerMetadata{
-		AuthorizationEndpoint:                      oauth2BaseURL.JoinPath("authorize").String(),
+	return &oauth.AuthorizationServerMetadata{
+		AuthorizationEndpoint:                      "openid4vp:",
 		ClientIdSchemesSupported:                   clientIdSchemesSupported,
 		DPoPSigningAlgValuesSupported:              jwx.SupportedAlgorithmsAsStrings(),
 		GrantTypesSupported:                        grantTypesSupported,
-		Issuer:                                     identity.String(),
+		Issuer:                                     ownedDID.String(), // todo: according to RFC8414 this should be a URL starting with https
```

Comment:
for wallets, shouldn't this be SIOPv2 in certain cases? Or don't we ever do SIOPv2 currently?

(my question is probably: how does this relate)

Reply:
[we don't use SIOP(v2), only OpenID4VP](https://openid.net/specs/openid-connect-self-issued-v2-1_0.html#name-presentation-of-claims-from). Claims should come from trusted issuers, not be self attested.